src/lib/snarky_group_map/group_map.ml

```diff
@@ -55,378 +23,390 @@ struct
-         ; ((3, 2), "-4488")
-         ; ((0, 4), "-5")
-         ; ((6, 1), "2108304")
-         ; ((3, 3), "2904")
-         ; ((0, 5), "1") ])
+  include T
+  include Hashable.Make (T)
+end
 
-  let n2tbl =
-    Hashtbl.of_alist_exn
-      (module Hash_key)
-      (field_map
-         [ ((0, 0), "-1")
-         ; ((0, 1), "6")
-         ; ((3, 0), "-4356")
-         ; ((0, 2), "-15")
-         ; ((3, 1), "-424944")
-         ; ((0, 3), "20")
-         ; ((6, 0), "-6324912")
-         ; ((3, 2), "-26136")
-         ; ((0, 4), "-15")
-         ; ((6, 1), "12649824")
-         ; ((3, 3), "17424")
-         ; ((0, 5), "6")
-         ; ((9, 0), "-3061257408")
-         ; ((6, 2), "-6324912")
-         ; ((3, 4), "-4356")
-         ; ((0, 6), "-6") ])
+let interval x y = Sequence.range ~start:`inclusive ~stop:`inclusive x y
 
-  let d2tbl =
-    Hashtbl.of_alist_exn
-      (module Hash_key)
-      (field_map
-         [ ((0, 0), "1")
-         ; ((0, 1), "-4")
-         ; ((3, 0), "5976")
-         ; ((0, 2), "6")
-         ; ((3, 1), "-5808")
-         ; ((0, 3), "-4")
-         ; ((6, 0), "2108304")
-         ; ((3, 2), "2904")
-         ; ((0, 4), "1") ])
+(* for v, k, powers produces v^0, v^1, ... v^k *)
+let powers ~mul ~one v k =
+  let len = k + 1 in
+  let arr = Array.create ~len one in
+  let rec go acc i =
+    arr.(i) <- acc ;
+    let i = i + 1 in
+    if i < len then go (mul acc v) i
+  in
+  go v 1 ; arr
+
+module Params = struct
+  type 'f coefficients =
+    {n1: 'f; d1: 'f; n2: 'f; d2: 'f; n3: 'f; d31: 'f; d32: 'f}
+
+  let map {n1; d1; n2; d2; n3; d31; d32} ~f =
+    {n1= f n1; d1= f d1; n2= f n2; d2= f d2; n3= f n3; d31= f d31; d32= f d32}
+
+  let t =
+    { n1=
+        AB_pair.Table.of_alist_exn
+          [ ((0, 0), "212")
+          ; ((0, 1), "-208")
+          ; ((3, 0), "-161568")
+          ; ((0, 2), "-264")
+          ; ((3, 1), "441408")
+          ; ((0, 3), "304")
+          ; ((6, 0), "-92765376")
+          ; ((3, 2), "-127776")
+          ; ((0, 4), "-44") ]
+    ; d1=
+        AB_pair.Table.of_alist_exn
+          [ ((0, 0), "-1")
+          ; ((0, 1), "5")
+          ; ((3, 0), "10536")
+          ; ((0, 2), "-10")
+          ; ((3, 1), "9480")
+          ; ((0, 3), "10")
+          ; ((6, 0), "4024944")
+          ; ((3, 2), "-4488")
+          ; ((0, 4), "-5")
+          ; ((6, 1), "2108304")
+          ; ((3, 3), "2904")
+          ; ((0, 5), "1") ]
+    ; n2=
+        AB_pair.Table.of_alist_exn
+          [ ((0, 0), "-1")
+          ; ((0, 1), "6")
+          ; ((3, 0), "-4356")
+          ; ((0, 2), "-15")
+          ; ((3, 1), "-424944")
+          ; ((0, 3), "20")
+          ; ((6, 0), "-6324912")
+          ; ((3, 2), "-26136")
+          ; ((0, 4), "-15")
+          ; ((6, 1), "12649824")
+          ; ((3, 3), "17424")
+          ; ((0, 5), "6")
+          ; ((9, 0), "-3061257408")
+          ; ((6, 2), "-6324912")
+          ; ((3, 4), "-4356")
+          ; ((0, 6), "-6") ]
```

Comment:
this should be `-1` (this is my fault! I didn't trust myself to double check it properly last time after staring at so many numbers so I double checked all the numbers this time)